Added real world projects

K92hoang opened the pull request:
- Added python-string-utils project, this project has string typ hints
- Added flutes project (e. g. in "An empirical study of automated unit test generation
for Python" Paper Page 32)

K92hoang committed:
Added 2 real world projects

Lexolef reviewed (approved):
Nice looks good for testing. I am not sure whether it is necessary for us to fix the ruff errors here. 😅

K92hoang committed:
Fixed ruff errors

Lexolef reviewed (approved):
Looks fine now. I added a screenshot of the last ruff message I got. But I approve anyway 😄

K92hoang committed:
Fixed missing ruff error

Priapos1004 committed:
Revert "Fixed missing ruff error"

This reverts commit dbe205910a548a151cda6ae57bbfa0e5e2ee8c20.

Priapos1004 commented:
Reverted changes made for ruff, as the flag ruff error was because of an older version of ruff that Levin used

Priapos1004 committed:
Removed flutes folder

- Only the management files were copied and also the other files have nearly no pure string type functions

Priapos1004 committed:
Fixed benchmarking for string_utils folder

Priapos1004 committed:
Merge branch 'main' into real_world_test

Priapos1004 committed:
Added random seed

Priapos1004 committed:
Merge branch 'main' into real_world_test

K92hoang committed:
Created one folder for all self made tests

Priapos1004 committed:
Added Benchmark Repo

- Changed timing from seeding log to4 decimals (before: 2 decimals)
- Excluded new repo from ruff
- Added timing log for run_....py

Priapos1004 committed:
Revert "Created one folder for all self made tests"

This reverts commit b113b0e9f683d6224d3fdd57248c3a6d36551844.

Priapos1004 commented:
Added TheAlgorithm repository as benchmark repo. TO-DO: Automated benchmark run for multiple seeds and creation of one csv file (I will try to do this till noon)

Priapos1004 committed:
Added automated running of multiple seeds and strategies

Priapos1004 committed:
Added automated csv / excel creation of merged reports

Priapos1004 committed:
Restructured mixed_funcs

- Corrected spelling in config.py

Priapos1004 reviewed (approved):
Should be fine now
